content/ru/ebpf.md

```diff
@@ -0,0 +1,45 @@
+---
+
+eBPF (англ. extended Berkley Packet Filter — расширенный фильтр пакетов Беркли) это технология,
+которая позволяет небольшим программам или скриптам запускаться в пространстве ядра ОС Линукс 
+без необходимости изменять код ядра или загружать модули ядра.
+
+ОС Линукс имеет два пространства: пространство ядра (kernel space) и пользовательское (user space). 
+Ядро представляет основу операционной системы и имеет неограниченный доступ к аппаратному обеспечению.
+
+Приложения находятся в пользовательском пространстве. 
+Когда возникает необходимость в привилегированном доступе, приложения отправляют запрос к ядру.
+Для приложений, требующих большей свободы действий, например, для прямого доступа к аппаратному обеспечению, 
+ядро может быть расширено за счёт моделей ядра Линукс. 
+Такой подход расширяет стандартные функции ядра, давая приложениям доступ к низкоуровневым компонентам.
+Однако расширение ядра за счет модулей влечет за собой риски безопасности, делая eBPF привлекательной альтернативой.
+
+## Какую проблему решает
+
+Как правило, приложения работают в пользовательском пространстве,
+и если возникает необходимость в привилегированном доступе (например, доступ к аппаратному обеспечению),
+приложение отправляет запрос к ядру также известный как "системный вызов".
+В большинстве случает такой подход прекрасно работает. 
+Однако бывают ситуации, когда разработчики нуждаются в более низкоуровневом доступе к системе.
+Такие ситуации, как правило, связаны с наблюдаемостью (observability), безопасностью и работой с сетью.
+Использование модулей ядра Линукс позволяет решить вышеупомянутую проблему без изменения исходного кода ядра.
+Но несмотря на преимущества, такой подход влечет за собой риски безопасности.
+Код модулей ядра работает в пространстве ядра, а это значит, что сбой модуля неминуемо приводит к сбою всего ядра и как следствие — сбою всей аппаратной системы.
+Вдобавок, модули ядра имеют повышенные привилегии и прямой доступ к системным ресурсам. Без должной безопасности, они могут стать источником уязвимостей для злоумышленников.
+
+## Как именно решает проблему
+
+По сравнению с модулями ядра, eBPF предоставляет более контролируемую и ограниченную среду для исполнения пользовательских программ.
+eBPF программа запускается в "песочнице" внутри ядра, обеспечивая тем самым изоляцию и снижая риски.
+Если в eBPF программе обнаружится уязвимость или недочет, то их воздействие будет ограничено "песочницей".
+Более того, прежде чем программа сможет запуститься внутри ядра, она должна пройти процесс верификации.
+Верификатор проверяет eBPF программу на наличие потенциально небезопасного кода, 
+такого как доступ к памяти за пределами выделенного буфера, бесконечные циклы и неавторизованный доступ к функциям ядра.
+Верификация гарантирует что eBPF программа не войдет в бесконечный цикл и не приведет к сбою ядра.
+Эти меры безопасности делают eBPF более надежным вариантом запуска приложений в ядре, по сравнению с модулями ядра Линукс.
```

Comment:
```suggestion
Верификация гарантирует, что eBPF-программа не войдет в бесконечный цикл и не приведет к сбою ядра.
Такие меры безопасности делают eBPF более надежным вариантом запуска приложений в ядре по сравнению с модулями ядра Линукс.
```

Reply:
By the way, feel free to join our [Russian Cloud Native](https://t.me/ru_cloudnative) localizer community. 😊